src/AggregateRoots/AggregateRoot.php

```diff
@@ -43,6 +43,21 @@ public static function retrieve(string $uuid)
         return $aggregateRoot->reconstituteFromEvents();
     }
 
+    /**
+     * @param string $uuid
+     * @param \DateTimeImmutable $tillDateTime
+     *
+     * @return static
+     */
+    public static function retrieveUntil(string $uuid, \DateTimeImmutable $tillDateTime)
```

Comment:
Use the `self` type hint, remove the docblock, and import all namespaces so that `\` can be removed.